src/libbson/tests/test-utf8.c

```diff
@@ -81,6 +85,47 @@ test_bson_utf8_escape_for_json (void)
    str = bson_utf8_escape_for_json (unescaped, -1);
    BSON_ASSERT (0 == memcmp (str, "\\u000e", 7));
    bson_free (str);
+
+   // Invalid UTF-8 strings (should return null)
+
+   // 2 bytes expected
+   {
+      str = bson_utf8_escape_for_json ("\xc2", -1);
+      BSON_ASSERT (!str);
```

Comment:
```suggestion
      ASSERT (!str);
```
Use `ASSERT` in tests instead of `BSON_ASSERT`.

Drive-by update to other assertions in this function/file to use `ASSERT` is optional.